### PR TITLE
Options to plain json (clear)

### PR DESCRIPTION
**Pull request by dvpolyakov**
Implemented OptionsToPlainJson function that generates plain JSON with all parameters from documentation set in training from trainingOptions and outputFilesOptions. Unit tests are added. Python interface will be added. Finally it will resolve https://github.com/catboost/catboost/issues/833

I hereby agree to the terms of the CLA available at: [link](https://yandex.ru/legal/cla/?lang=ru).

### CODE REVIEW

**Comment by annaveronika** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@annaveronika, internal review request created: [859098](https://a.yandex-team.ru/review/859098)

**Comment by arcadia-devtools** [comment]
Check [859098](https://a.yandex-team.ru/review/859098) failed: 12 test(s) broken

**Comment by annaveronika** [comment]
catboost/python-package/ut/medium - these tests are failing.


MEDIUM TEST | catboost/python-package/ut/medium (test.py::test_onnx_export[binclass])

Test results differ from canonical: files content differs: ''https://proxy.sandbox.yandex-team.ru/1007448001/build-release/catboost.tar/catboost/tools/model_comparator/model_comparator --diff-limit 0' has finished unexpectedly with rc = 1
stderr: ONNX models differ

File: catboost/python-package/catboost/core.py [hunk 1]
```diff
@@ -2120,6 +2123,21 @@ def get_params(self, deep=True):
         else:
             return params
 
+    def get_all_params(self, deep=True):
```
Comment (annaveronika) [inline catboost/python-package/catboost/core.py]: зачем здесь deep?

File: catboost/python-package/catboost/helpers.cpp [hunk 2]
```diff
@@ -166,3 +168,18 @@ NJson::TJsonValue GetTrainingOptions(
     catboostOptions.Save(&catboostOptionsJson);
     return catboostOptionsJson;
 }
+
+NJson::TJsonValue GetPlainJsonWithAllOptions(const TFullModel& model, bool categoricalFeaturesArePresent)
```
Comment (annaveronika) [inline catboost/python-package/catboost/helpers.cpp]: categoricalFeaturesArePresent => hasCatFeatures

File: catboost/python-package/catboost/helpers.cpp [hunk 3]
```diff
@@ -166,3 +168,18 @@ NJson::TJsonValue GetTrainingOptions(
...
+{
+    NJson::TJsonValue trainOptions = ReadTJsonValue(model.ModelInfo.at("params"));
+    NJson::TJsonValue outputOptions = ReadTJsonValue(model.ModelInfo.at("output_options"));
+    NJson::TJsonValue plainOptionsJson;
+    NCatboostOptions::ConvertOptionsToPlainJson(trainOptions, outputOptions, &plainOptionsJson);
+    CB_ENSURE(!plainOptionsJson.GetMapSafe().empty(), "plainOptionsJson should not be empty.");
+    NJson::TJsonValue plainOptionsJsonEfficient(plainOptionsJson);
+    CB_ENSURE(!plainOptionsJsonEfficient.GetMapSafe().empty(), "problems with copy constructor.");
+    NCatboostOptions::DeleteEmptyKeysInPlainJson(&plainOptionsJsonEfficient, categoricalFeaturesArePresent);
+    CB_ENSURE(!plainOptionsJsonEfficient.GetMapSafe().empty(), "plainOptionsJsonEfficient should not be empty.");
```
Comment (annaveronika) [inline catboost/python-package/catboost/helpers.cpp]: plainOptionsJson => plainOptions (no obvious need to write type here)
plainOptionsJsonEfficients => cleanedOptions

File: catboost/libs/options/plain_options_helper.cpp [hunk 4]
```diff
@@ -44,6 +48,26 @@ static Y_NO_INLINE void CopyCtrDescription(
     seenKeys->insert(TString(srcKey));
 }
 
+static Y_NO_INLINE void ConcatenateCtrDescription(
```
Comment (annaveronika) [inline catboost/libs/options/plain_options_helper.cpp]: метод должен делать только то, что указано в его названии. Если написано - ConcatenateCtrDescription, то это он и должен делать. Добавлять при этом в дополнительный сет seenKeys этот метод не должен, это нужно делать отдельно.

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request updated: [859098](https://a.yandex-team.ru/review/859098)

**Comment by arcadia-devtools** [comment]
Check [859098](https://a.yandex-team.ru/review/859098) failed: 18 test(s) broken

**Pull request merged**